test/test_execution.py

```diff
@@ -215,39 +220,70 @@ def runner() -> int:
-            print(f"Attempt {i} failed with msg: {e}", file=sys.stderr)
-            sleep(0.1)  # Retry after a short delay
-
-    if pid is not None:
-        os.kill(pid, signal.SIGTERM)
+    wait_time = 0.3
+    proc = multiprocessing.Process(target=runner)
+    proc.start()
+    sleep(wait_time)
+    assert proc.pid is not None, "Process PID should not be None"  # for mypy
+    os.kill(proc.pid, signal.SIGINT)
+    proc.join()
+
+    # Once the command has been killed, duct should exit gracefully with exit code 0
+    assert proc.exitcode == 0
+
+    if fail_time is None or fail_time != 0:
+        assert_expected_files(temp_output_dir, exists=False)
     else:
-        raise RuntimeError("Failed to find sleep process")
+        # proc exit code should Cannot retrieve the exit code from the thread, it is written to the file
+        with open(os.path.join(temp_output_dir, SUFFIXES["info"])) as info:
+            info_data = json.loads(info.read())
+
+        command_exit_code = info_data["execution_summary"]["exit_code"]
+        # SIGINT
+        assert command_exit_code == 128 + 2
 
-    thread.join()
+
+@pytest.mark.parametrize("fail_time", [None, 0, 10, -1, -3.14])
+def test_signal_kill(temp_output_dir: str, fail_time: float | None) -> None:
+
+    def runner() -> int:
+        script_path = str(TEST_SCRIPT_DIR / "signal_ignorer.py")
+        kws = {}
+        if fail_time is not None:
+            kws["fail_time"] = fail_time
+        args = Arguments.from_argv([script_path], output_prefix=temp_output_dir, **kws)
+        return execute(args)
+
+    wait_time = 0.6
+    proc = multiprocessing.Process(target=runner)
+    proc.start()
+    sleep(wait_time)
+    assert proc.pid is not None, "Process PID should not be None"  # for mypy
+    os.kill(proc.pid, signal.SIGINT)
+    sleep(wait_time)
+    os.kill(proc.pid, signal.SIGINT)
+    sleep(wait_time)
+    os.kill(proc.pid, signal.SIGINT)
+    proc.join()
+
+    # Once the command has been killed, duct should exit gracefully with exit code 0
+    assert proc.exitcode == 0
 
     if fail_time is None or fail_time != 0:
         assert_expected_files(temp_output_dir, exists=False)
     else:
-        # Cannot retrieve the exit code from the thread, it is written to the file
+        # Cannot retrieve the command exit code from the thread, get from duct log
         with open(os.path.join(temp_output_dir, SUFFIXES["info"])) as info:
             info_data = json.loads(info.read())
 
-        exit_code = info_data["execution_summary"]["exit_code"]
-        assert exit_code == 128 + 15
+        command_exit_code = info_data["execution_summary"]["exit_code"]
+        # SIGKILL
+        assert command_exit_code == 128 + 9
 
 
 def test_duct_as_executable(temp_output_dir: str) -> None:
-    ps_command = f"{sys.executable} {__main__.__file__} -p {temp_output_dir} sleep 0.01"
+    ps_command = (
+        f"{sys.executable} {__main__.__file__} -p {temp_output_dir} -q sleep 0.01"
```

Comment:
since you are touching this, needs quotes around arguments or a `list` representation used so that if there is a space in any path - does not crash . Best would be a list form if possible here

Reply:
I don't have any further comments atm , IIRC it worked when I tried, so - let' proceed overall